packages/autorest.go/src/generator/interfaces.ts

```diff
@@ -33,6 +33,9 @@ export async function generateInterfaces(session: Session<CodeModel>): Promise<s
     }
   }
 
+  // remove duplicate
+  allDiscriminators = Array.from(new Set(allDiscriminators));
```

Comment:
Maybe we should just update the condition on line 30 to skip adding an entry if it already exists.

Also, do you have an example of where this happens?

Reply:
Encountered when generating services: 
- sdk/resourcemanager/security/armsecurity
- sdk/resourcemanager/mediaservices/armmediaservices